### PR TITLE
Update User32::WinEventProc and MSCorEE::GetVersionFromProcess

### PR DESCRIPTION
**Pull request by joerohde**
User32: Update WinEventProc - Runtime Issue
- Param 1 is IntPtr - Income parm should be be SafeEvent...
MSCorEE: Update GetVersionFromProcess - Compile Issue
- Can't cast [] to * without marking method unsafe and using a 'fixed' scope block.

### CODE REVIEW

File: src/User32/User32.cs [hunk 1]
```diff
@@ -162,7 +162,7 @@ public static partial class User32
         /// <param name="dwmsEventTime">Specifies the time, in milliseconds, that the event was generated.</param>
         [UnmanagedFunctionPointer(CallingConvention.StdCall)]
         public delegate void WinEventProc(
-            SafeEventHookHandle hWinEventHook,
+            IntPtr hWinEventHook,
```
Comment (AArnott) [inline src/User32/User32.cs:165]: Why are we switching from a SafeHandle to an IntPtr here? From the docs, it seems this value comes from `SetWinEventHook`, which (still) returns a `SafeEventHookHandle`, so isn't it both easier and safer to keep this as a `SafeEventHookHandle` here as well?

File: src/MSCorEE/MSCorEE.Helpers.cs [hunk 2]
```diff
@@ -73,14 +73,19 @@ public static unsafe byte[] StrongNameTokenFromPublicKey(byte[] publicKeyBlob)
...
             while (true)
             {
                 int dwLength;
-                HResult hr = GetVersionFromProcess(hProcess, versionChars, versionChars.Length, out dwLength);
+                HResult hr = -1;
+                fixed (char* versionCharsPtr = versionChars)
+                {
+                    hr = GetVersionFromProcess(hProcess, versionCharsPtr, versionChars.Length, out dwLength);
+                }
+
```
Comment (AArnott) [inline src/MSCorEE/MSCorEE.Helpers.cs]: Why does this change need to be made? There is already an overload of `GetVersionFromProcess` that basically does this that we were calling before you made this change:

```cs
        public static unsafe HResult GetVersionFromProcess(SafeHandle hProcess, char[] pVersion, int cchBuffer, out int dwLength)
        {
            fixed (char *pVersionLocal = pVersion)
            {
                HResult result = GetVersionFromProcess(hProcess, pVersionLocal, cchBuffer, out dwLength);
                return result;
            }
        }
```

This overload is generated during the build, so you wouldn't see it in checked in source code, but in VS after building once you can Go To Definition on this call and see the generated code.

**Review by AArnott — CHANGES_REQUESTED** [review]
I don't understand why either of these changes are needed. Can you elaborate?

File: src/User32/User32.cs (hunk 1 above)
Comment (vbfox) [reply src/User32/User32.cs:165]: It produce a runtime error, the marshaller currently doesn't handle theses

> 'Cannot marshal 'parameter #1': SafeHandles cannot be marshaled from unmanaged to managed.'

Still seem to be that case in coreclr https://github.com/dotnet/coreclr/blob/2b50bba8131acca2ab535e144796941ad93487b7/src/vm/ilmarshalers.cpp#L2866 I don't know what make this case more special than an out parameter but well it's like that for now

**Review by AArnott — APPROVED** [review]
I reverted the unnecessary change and will take the rest. Thanks.

File: src/MSCorEE/MSCorEE.Helpers.cs (hunk 2 above)
Comment (AArnott) [reply src/MSCorEE/MSCorEE.Helpers.cs]: No reason I guess. I'm reverting this change.

**Review by AArnott — APPROVED** [review]

**Comment by FDB-Fakim** [comment]
Is there a reason https://github.com/AArnott/pinvoke/commit/9476c5cc9d067f78b45e7565701059286b9ac42a isn't included in PInvoke.User32.0.6.6.nupkg?

**Comment by AArnott** [comment]
@FDB-Fakim Yes: that release came out before this PR completed. :)

**Comment by AArnott** [comment]
I think what you mean to ask is "Is there a reason there hasn't been a release since this PR was merged?"

And the answer to that would be "not particularly." I suppose it is time to release another one. I'll prepare one.

**Pull request merged**